.github/CHANGELOG.md

```diff
@@ -2,6 +2,66 @@
 
 <h3>New features since last release</h3>
 
+* Adds a decorator `@qml.qfunc_transform` to easily create a transformation
+  that modifies the behaviour of a quantum function.
+  [(#1315)](https://github.com/PennyLaneAI/pennylane/pull/1315)
+
+  For example, consider the following transform, which replaces all
+  `CRX` gates with a sequence of `RX`, `RY`, and `CZ` gates:
+
+  ```python
+  @qml.qfunc_transform
```

Comment:
Why does it take a `tape` if it's a "qfunc" transform?  

I thought "qfunc" transforms took in quantum functions and returned quantum functions?

Reply:
@albi3ro that's exactly right! You could always write a qfunc transform manually like so, without ever needing to use any decorators:

```python
def transform(old_qfunc, params):
    def new_qfunc(*args, **kwargs):
        # 1. extract the tape from the old qfunc, being
        # careful *not* to have it queued.
        tape = make_tape(old_qfunc)(*args, **kwargs)

        # 2. transform the tape
        new_tape = tape_transform(tape, params)

        # 3. queue the *new* tape to the active queuing context
        new_tape.queue()
    return new_qfunc
```

Note that this is pseudocode - steps (1) and (3) are much more complicated in practice than they are presented here! It's actually quite non-trivial to get the above working in general, because it requires:

- an intimate knowledge of the queuing system and how it works

- how to convert qfuncs into tapes without them being queued

- dealing with potentially nested queuing contexts

- having to take into account that qfuncs can be called in lots of different places

Steps (1) and (3) are actually the same no matter the qfunc transform; really, it is only step (2), `tape_transform` and the corresponding tape transform parameters, that really define the qfunc transformation.

So the idea with the decorator is therefore:

- Allow a dev to provide a tape transform that *defines the intended qfunc transform*
- All the queuing boilerplate (steps 1 and 3) are added *automatically* by the decorator (and are also tested thoroughly by the decorator tests, reducing maintenance overhead and chances of bugs!)


In essence, you can think of the decorator as taking a `tape_transform`, and _elevating it_ into a qfunc transform.